internal/pkg/cli/local_run.go

```diff
@@ -292,23 +295,56 @@ func (o *localRunOpts) Execute() error {
 	}
 	o.imageInfoList = append(o.imageInfoList, sidecarImageLocations...)
 
-	err = o.runPauseContainer(context.Background(), containerPorts)
-	if err != nil {
-		return err
-	}
+	errCh := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Handle containers when a termination signal is received.
+	go func() {
+		sig := o.NotifyInterrupt()
+		log.Println("Received signal:", sig)
+		cancel()
+		errCh <- o.handleContainers()
```

Comment:
if this returns nil, then the program will exit before the below code runs, right?

Reply:
Yes you are right. Since we get at least nil to the error channel, we exit the execute function.